SPECS/bcc+clang.spec

```diff
@@ -57,7 +57,7 @@ make install/strip DESTDIR=%{buildroot}
 
 %package -n libbcc
 Summary: Shared Library for BPF Compiler Collection (BCC)
-Requires: elfutils-libelf
+Requires: elfutils-libelf elfutils-debuginfod-client
```

Comment:
I am not a rpm build expert either. But I think we can do something like
```
having_debuginfod = %{?debuginfod 1 }  %{!?debuginfod 0} 
%if having_debuginfod == 1
Requires: elfutils-debuginfod-client
%endif
```

And in bcc/scripts/ various build scripts, you can detect whether debuginfod is available or not, if available
add `--define "debuginfod 1"` to the rpmbuild command line.

Reply:
Another option would be to do something like this in `bcc.spec`:
```
# Build with debuginfod support for Fedora >= 32
%if 0%{?fedora} >= 28
%bcond_with libdebuginfod
%else
%bcond_without libdebuginfod
%endif
[...]
%if %{with libdebuginfod}
BuildRequires: elfutils-debuginfod-client-devel
%endif
[...]
%package -n libbcc
Summary: Shared Library for BPF Compiler Collection (BCC)
Requires: elfutils-libelf
%if %{with_libdebuginfod}
Requires: elfutils-debuginfod-client
%endif
```

I'm not familiar with the build process for the packages, are `SPECS/bcc+clang.spec` and `scripts/build-release-rpm.sh` used at all? It seems that most variable parts are in `bcc.spec` which is used by `build-rpm.sh`.

Unfortunately, `debian/control` does not have any similar options to dynamically modify the package build process, so we would need some kind of template as a base and replace the options from `build-deb.sh`, if I understand correctly.

Reply:
> I'm not familiar with the build process for the packages, are SPECS/bcc+clang.spec and scripts/build-release-rpm.sh used at all? It seems that most variable parts are in bcc.spec which is used by build-rpm.sh.
I think buildbot may use them. Fedora may use different ways to build, I suspect.

For debian/control, not sure who used it. May be fine to leave it, but some comments in the file will be still good.